### PR TITLE
Docker registry auth

### PR DESCRIPTION
**Pull request by priteau**

### CODE REVIEW

**Comment by mnasiadka** [comment]
What about https://github.com/stackhpc/ansible-collection-hashicorp/pull/12?

**Review by markgoddard — APPROVED** [review]

**Comment by markgoddard** [comment]
> What about #12?

I prefer this approach

**Comment by mnasiadka** [comment]
> > What about #12?
> 
> I prefer this approach

Fine by me, closed the other one.

File: roles/vault/tasks/main.yml [hunk 1]
```diff
@@ -2,6 +2,9 @@
 - name: "Vault Prechecks"
   import_tasks: prechecks.yml
 
+- name: "Vault Prerequisites"
+  import_tasks: prereqs.yml
```
Comment (mnasiadka) [inline roles/vault/tasks/main.yml:6]: Do we really need that in a separate file? It's not really a prerequisite.

Comment (markgoddard) [reply roles/vault/tasks/main.yml:6]: It is, isn't it?

Comment (mnasiadka) [reply roles/vault/tasks/main.yml:6]: Well, maybe it is, but does it deserve a separate file? If we would have more - sure.

Comment (priteau) [reply roles/vault/tasks/main.yml:6]: I asked myself the same question, but it's not a prechecks and we need it for both consul and vault. I could have added it at the beginning of the consul play, but separating it seemed more logical.

**Pull request merged**